src/MolSim.cpp

```diff
@@ -73,32 +159,69 @@ int main(int argc, char *argsv[]) {
 }
 
 void calculateF() {
-  std::list<Particle>::iterator iterator;
-  iterator = particles.begin();
-
   for (auto &p1 : particles) {
+    std::array<double, 3> force = {0,0,0};
     for (auto &p2 : particles) {
-      // @TODO: insert calculation of forces here!
+      if (p1 == p2) {
+        continue;
+      }
+
+      // F_{ij} = m_i m_j (||x_i - x_j||_2)^{-3} (x_j - x_i)
+      // only (x_j - x_i) changes for each vector entry
+      // factor calculates m_i m_j (||x_i - x_j||_2)^{-3}
+      double factor = 0.0;
+      for (int i = 0; i < 3; ++i) {
+        const double diff = (p1.getX()[i] - p2.getX()[i]);
```

Comment:
As pointed out in the meeting today, you might wanna have a look at `ArrayUtils.h`, especially `L2 norm` might be helpful.